File: Difficulty: Easy/Trailing zeroes in factorial/trailing-zeroes-in-factorial.py
#User function Template for python3

class Solution:
    def trailingZeroes(self, N):
        count = 0
        # Keep dividing N by powers of 5 and add the quotient to the count
        while N >= 5:
            N //= 5
            count += N
        return count



#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int(input())
    for _ in range(t):
        N = int(input()) 
        ob = Solution()
        ans = ob.trailingZeroes(N)
        print(ans)
# } Driver Code Ends

File: Difficulty: Easy/Trailing zeroes in factorial/trailing-zeroes-in-factorial.cpp
//{ Driver Code Starts
#include<bits/stdc++.h> 
using namespace std; 

// } Driver Code Ends
//User function Template for C++
class Solution
{
    public:
    int trailingZeroes(int N) {
        int count = 0;
        
        // Keep dividing N by powers of 5 and update the count
        for (int i = 5; N / i > 0; i *= 5) {
            count += N / i;
        }
        
        return count;
    }
};

//{ Driver Code Starts.
int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int N;
        cin >> N;
        Solution ob;
        int ans  = ob.trailingZeroes(N);
        cout<<ans<<endl;
    }
    return 0;
}
// } Driver Code Ends